File: hjkim/Week3/01.17/PGS_의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> map = new HashMap<>();
        for (String[] cloth : clothes) {
            String type = cloth[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }

        for (Integer value : map.values()) {
            answer *= value + 1;
        }
        return answer - 1;
    }
}

File: raeunlee/Week3/01.17/PGS_의상.py
def solution(clothes):   
    # 옷장 딕셔너리
    closet = {} 
    
    # 종류를 key로 둔다
    for name, kind in clothes:
        if kind in closet.keys():
            closet[kind] += [name]
        else:
            closet[kind] = [name]

    answer = 1
    # A만 사용, B만 사용, AB 사용, 1 모두 안사용 -> (A+1)(B+1) = AB + A + B + 1
    # 종류, 의상
    for k , value in closet.items():
        answer *= (len(value) + 1)
    
    return answer - 1 # 아무것도 입지 않은 경우 1 제외

